app/services/company.js

```diff
@@ -1,56 +1,83 @@
 const path = require('path');
 
-const companyFilesFolder = require('../../config.js').companyFilesFolder;
 const addressRepository = require('../repositories/address');
 const fieldOfActivityRepository = require('../repositories/fieldOfActivity');
 const fileRepository = require('../repositories/file');
+const companyValidator = require('../validators/company');
+const addressService = require('../services/address');
 const timetableService = require('./timetable');
 const fileService = require('./file');
 
 const error = require('./error');
 const CrudService = require('./crudService');
+const { COMPANY_FILES_FOLDER, ORIGIN } = require('../../config.js');
 
 class CompanyService extends CrudService {
-  async create(company, files) {
-    if (company.id !== undefined) {
-      error.throwValidationError('Invalid company format.');
+  async get(id) {
+    const company = await this.repository.get(id);
+    if (!company) {
+      error.throwNotFoundError('Company not found.');
     }
 
-    const companyExists = await this.repository.getByPropsNonParanoid({ name: company.name });
-    if (companyExists) {
-      error.throwValidationError('Company already exists.');
+    const { fieldOfActivityId, ...companyToReturn } = company;
+    const fieldOfActivity = await fieldOfActivityRepository
+      .get(fieldOfActivityId);
+    companyToReturn.business = fieldOfActivity.name;
+
+    const companyAddresses = await addressRepository.getAllByProps({
+      companyId: id,
+    });
+    if (companyAddresses && companyAddresses.length) {
+      companyToReturn.address = companyAddresses[0];
     }
 
+    const companyFiles = await fileRepository.getAllByProps({
+      entityId: id,
+      entityType: 'company',
+    });
+    companyToReturn.images = companyFiles.map(file => ({
+      name: file.originalFileName,
+      url: `${ORIGIN}companies/${file.originalFileName}`,
+    }));
+
+    return companyToReturn;
+  }
+
+  async create(company, files) {
+    await companyValidator.validateCreate(company);
+
     const {
       business,
+      address,
       timetables,
       ...newCompany
     } = company;
-    if (business !== undefined) {
-      const fieldOfActivity = await fieldOfActivityRepository.getByPropsNonParanoid({ name: business });
-      if (!fieldOfActivity) {
-        error.throwNotFoundError('Field of activity not found.');
-      }
-      newCompany.fieldOfActivityId = fieldOfActivity.id;
-    } else {
-      error.throwValidationError('Invalid company format.');
-    }
+    const fieldOfActivity = await fieldOfActivityRepository.getByPropsNonParanoid({
+      name: business,
```

Comment:
You should have sent the field of activity id from the frontend. Keep this in mind in the future, don't change anything here now.